Exposed TO_EXP_NEG and TO_EXP_POS.

loriopatrick opened the pull request:
Exposed TO_EXP_NEG and TO_EXP_POS like DP and RM so you can control output of Big.toString()

MikeMcl commented on big.js:
```diff
@@ -986,8 +988,8 @@
+     * or greater than Big.TO_EXP_POS, or a negative exponent equal to or less than
+     * Big.TO_EXP_NEG.
      */
     P.toString = P.valueOf = P.toJSON = function () {
```
`var Big = x.constructor;` would need to be included before this (see the `format` function for example), otherwise `Big` will refer always to the original `Big` constructor.

MikeMcl commented on big.js:
```diff
@@ -1067,11 +1069,11 @@
             x = this,
-            neg = TO_EXP_NEG,
-            pos = TO_EXP_POS;
+            neg = Big.TO_EXP_NEG,
```
As before, `Big = x.constructor,` would need to be included before this, otherwise `Big` will refer always to the original `Big` constructor.

MikeMcl commented:
Looked good, but won't work correctly if multiple Big constructors are defined. For example,

    var Big2 = Big();
    Big2.TO_EXP_POS = 1;
    var x = new Big2(123.456);
    x.toString();    // ?

loriopatrick commented:
I've renamed the variables to be E_POS and E_NEG like you suggested and got it work with multiple constructors. I also updated the binary, docs, changelog, and version number. Let me know if I missed anything.

MikeMcl commented:
I'll have a look at this properly over the weekend and get back to you after that.

MikeMcl commented on README.md:
```diff
@@ -186,6 +186,10 @@ See LICENCE.
 
 ## Change Log
 
+####3.0.3
```
I'd prefer the minor version number to be incremented, i.e. 3.1.0, as new functionality is being introduced.
"Exposed TO_EXP_NEG and TO_EXP_POS as Big.E_NEG and Big.E_POS."

MikeMcl commented on big.min.js:
```diff
@@ -1 +1 @@
-/* big.js v3.0.2 https://github.com/MikeMcl/big.js/LICENCE */(function(n){"use strict";function c(){function n(t){var i=this;if(!(i instanceof n))return t===void 0?c():new n(t);t instanceof n?(i.s=t.s,i.e=t.e,i.c=t.c.slice()):y(i,t);i.constructor=n}return n.prototype=t,n.DP=l,n.RM=a,n}function s(n,t,i){var e=n.constructor,r=t-(n=new e(n)).e,f=n.c;for(f.length>++t&&o(n,r,e.RM),f[0]?i?r=t:(f=n.c,r= ...
\ No newline at end of file
+(function(global){"use strict";var DP=20,RM=1,MAX_DP=1e6,MAX_POWER=1e6,E_NEG=-7,E_POS=21,P={},isValid=/^-?(\d+(\.\d*)?|\.\d+)(e[+-]?\d+)?$/i,Big;function bigFactory(){function Big(n){var x=this;if(!(x instanceof Big)){return n===void 0?bigFactory():new Big(n)}if(n instanceof Big){x.s=n.s;x.e=n.e;x.c=n.c.slice()}else{parse(x,n)}x.constructor=Big}Big.prototype=P;Big.DP=DP;Big.RM=RM;Big.E_NEG=E_NEG; ...
```
Please prepend:
/* big.js v3.1.0 https://github.com/MikeMcl/big.js/LICENCE */

MikeMcl commented on package.json:
```diff
@@ -1,7 +1,7 @@
   "name": "big.js",
   "description": "A small, fast, easy-to-use library for arbitrary-precision decimal arithmetic",
-  "version": "3.0.2",
+  "version": "3.0.3",
```
3.1.0

MikeMcl commented:
Also, please update the version numbers at line 1 and line 6 of the big.js file.

MikeMcl commented:
Thanks, Patrick.
